ros2/ubuntu/Dockerfile

```diff
@@ -1,44 +1,85 @@
-## Provide Orocos Toolchain (RTT/OCL) on top of the official ROS 2 docker images.
-ARG OROCOS_TOOLCHAIN_BRANCH=ros2
+## Provide Orocos Toolchain (RTT/OCL) and rtt_ros2_integration overlay workspace on top of the
+## official ROS 2 docker images.
+
 ARG ROS_DISTRO
 ARG UBUNTU_DISTRO
 
-## Stage deps: install additional Orocos Toolchain (with CORBA) dependencies
-FROM ros:${ROS_DISTRO}-ros-base-${UBUNTU_DISTRO} AS deps
-RUN apt-get update && apt-get install -y \
-    libboost-all-dev \
-    liblua5.1-0-dev \
-    libncurses5-dev \
-    libnetcdf-dev \
-    libreadline-dev \
-    libxml-xpath-perl \
-    omniorb omniidl omniorb-idl omniorb-nameserver libomniorb4-dev \
-    ruby bundler \
-    $([[ "$UBUNTU_DISTRO" != "focal" ]] && echo "ruby-facets") \
-    && rm -rf /var/lib/apt/lists/*
-
-## Stage build: clone and build orocos_toolchain with colcon
-FROM deps AS build
-ARG OROCOS_TOOLCHAIN_BRANCH
-RUN git clone --recursive https://github.com/orocos-toolchain/orocos_toolchain.git -b ${OROCOS_TOOLCHAIN_BRANCH} /build/orocos_toolchain
-RUN cd /build/orocos_toolchain \
-    && . /opt/ros/${ROS_DISTRO}/setup.sh \
-    && DESTDIR=/stage colcon build \
+## *************************************************************************************************
+## Stage base: prepare build environment
+## *************************************************************************************************
+FROM ros:${ROS_DISTRO}-ros-base-${UBUNTU_DISTRO} AS base
+
+# default config file for colcon
+COPY .colcon/defaults.yaml /root/.colcon/defaults.yaml
+
+# add custom rosdep rule file and update rosdep database
+COPY prereqs.yaml /etc/ros/rosdep/
+RUN echo "yaml file:///etc/ros/rosdep/prereqs.yaml" > /etc/ros/rosdep/sources.list.d/10-custom.list
+RUN rosdep update
+
+# setup new entrypoint and default command
+COPY orocos_entrypoint.sh /orocos_entrypoint.sh
+ENTRYPOINT [ "/orocos_entrypoint.sh" ]
+CMD [ "bash" ]
+
+## *************************************************************************************************
+## Stage orocos_toolchain: clone, build and install Orocos Toolchain using CMake (with CORBA)
+## *************************************************************************************************
+FROM base AS orocos_toolchain
+
+COPY orocos_toolchain.repos /build/
+RUN vcs import --recursive /build < /build/orocos_toolchain.repos \
+    && cd /build/orocos_toolchain \
+    && . /opt/ros/${ROS_DISTRO}/local_setup.sh \
+    #
+    # Install dependencies with rosdep
+    && apt-get update \
+    && apt-get install -y ruby \
+    && rosdep install -y --from-path . --ignore-src \
```

Comment:
This call in Ubuntu `focal` is still trying to install `ruby-facets`, maybe by one of its dependees `facets`.

Reply:
Fixed in 8ad33982ec7b1cc6ccdd5bbff4621fb1494f25be: use `gem` instead.
Debian/Ubuntu package [ruby-facets](https://packages.ubuntu.com/ruby-facets) as tracked by the [rosdep database](https://github.com/ros/rosdistro/blob/edd3289d06a3752bd8c988ae71091b490b0f160d/rosdep/ruby.yaml#L14) has been removed for some reason.